src/ipoplib.py

```diff
@@ -626,21 +747,46 @@ def parse_config():
     if not ("xmpp_username" in CONFIG and "xmpp_host" in CONFIG):
         raise ValueError("At least 'xmpp_username' and 'xmpp_host' must be "
                          "specified in config file or string")
-
+    # Use custom keyring incase of non-interactive environment
+    # (uses a fixed password)
+    if isinstance(keyring.get_keyring(),keyring.backends.file.EncryptedKeyring):
```

Comment:
I don’t understand your intention in this part. If we use custom keyring in non-interactive mode, shouldn’t the second argument “keyring.backends.file.PlaintextKeyring” instead of "keyring.backends.file.EncryptedKeyring"? I tested it in non-interactive mode, and it said PlaintextKeyring. DId I test it wrong?